x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichStore.java

```diff
@@ -53,7 +55,11 @@ public static void putPolicy(String name, EnrichPolicy policy, ClusterService cl
         if (name.toLowerCase(Locale.ROOT).equals(name) == false) {
             throw new IllegalArgumentException("Invalid policy name [" + name + "], must be lowercase");
         }
-        // TODO: add policy validation
+        Set<String> supportedPolicyTypes = Set.of(EnrichPolicy.SUPPORTED_POLICY_TYPES);
```

Comment:
Should we put this validation into the constructors of the EnrichPolicy instead?

Reply:
The reason I put it here, is because it makes generating random policies easier and
down the line I think that determining what types are valid isn't going to be checking
whether the matches with an entry in `EnrichPolicy.SUPPORTED_POLICY_TYPES`.